charts/spire/charts/spire-server/templates/configmap.yaml

```diff
@@ -31,58 +31,58 @@ server:
 
 plugins:
   DataStore:
-    - sql:
-        plugin_data:
-          database_type: "sqlite3"
-          connection_string: "/run/spire/data/datastore.sqlite3"
+    sql:
+      plugin_data:
+        database_type: "sqlite3"
+        connection_string: "/run/spire/data/datastore.sqlite3"
 
-  {{- with .Values.nodeAttestor.k8sPsat }}
+  {{- with .Values.nodeAttestor.k8s_psat }}
```

Comment:
Do we have to change the camelCase convention in values.yaml?

Reply:
I think point is about if we are to support camel case, snake case, or something else.  I'm going to put the need to document the policy in an issue; but, I'm basically calling for the following general rules:

1.  Proper names are to be represented in the case they come in, preserving all "Helm supported" characters per Helm guidelines, with unsupported / problematic characters replaced by underscores.  The rational is that it helps one identify the item, increasing usability.  (This replaces the hyphen '-' with underscores, making some items look like snake case; but they aren't snake case, for example freedom-FooFigthers would be freedom_FooFighters, not freedom_Foo_Fighters).

2. All other names are to be snake case, starting with a lower case letter, per Helm's documented general conventions.  

3.  All compound words including Acronyms (NASA, for National Aeronautics and Space Administration) and Initialisms (NFL, for National Football League) are to maintain an all-uppercase spelling, in accordance with the English language conventions.  If the camel case causes confusion "NASARAM", rename the word, ideally putting any still-existing Acronym or Initialism at the end "memoryNASA"

4. All compound words including abbreviations (shortening of words) will have the abbreviation in all lowecase, with the exception of when camel case indicates the abbreviation would be capitalized.  Examples include apptBuilding, destinationAppt, usersGroupsEtc.
 
5. If an abbreviation, initialisizm, or acronym would noromally be used with periods in the name (for example P.M.) we remove the periods without supplying an "alternate" character to maintain the spaceing (P.M. becomes PM, not P_M_ or P_M)

Exceptions to these guildelines should be rare; but are permitted, provided the current set of maintainers agree to violate the established guidelines.

(camelcase) 'k8sPsat' is wrong, because Psat isn't a word, it's an acronym, shortening "Projected Service Account Token", the appropriate name according to camelcase is 'k8sPSAT

FYi:
correct snakecase would be 'k8s_PSAT'.  Correct camel_case would be 'k8sPSAT' and correct "name case" would be "k8s_psat"

The last one is the one that most closely resembles the name of the node attestor plugin, we should use it.

Reply:
https://github.com/spiffe/helm-charts/issues/329

Reply:
If there are no objections, I'll resolve this conversation.  Until then, let's let 'k8s_psat' go forward, as there is only a convention to use camel case, and this seems the correct time to break the convention as it's referring to proper name (of the plugin).

Reply:
@marcofranssen apologies for resolving this conversation without hearing back from you, but a week has passed, so I am working under the assumption that you're ok with what I have said.  If you are not, please reopen the conversation.